### PR TITLE
DPC-754 add lookback failure reason to operation outcome

### PR DESCRIPTION
**Pull request by whuang85**
### Fixes [DPC-754](https://jira.cms.gov/browse/DPC-754)

Currently, we return an operation outcome when a patient fails look back, either because the patient had no EoBs or the patient's EoBs did not match the look back criteria.  Our current pilot partner would like more insight into why a patient's EoBs would not pass look back, therefore we are now adding which criteria the EoBs failed for look back into the operation outcome.

### Proposed Changes

Determine reason for look back failure by looking at the set of EoB's for a particular patient

### Change Details

* Determine reason for look back failure

### Security Implications

- [ ] new software dependencies

- [ ] security controls or supporting software altered

- [ ] new data stored or transmitted

- [ ] security checklist is completed for this change

- [ ] requires more information or team discussion to evaluate security implications

- [x] no PHI/PII is affected by this change

### CODE REVIEW

File: dpc-aggregation/src/main/java/gov/cms/dpc/aggregation/service/LookBackAnalyzer.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import org.hl7.fhir.dstu3.model.CodeableConcept;
+import org.hl7.fhir.dstu3.model.OperationOutcome;
+import org.hl7.fhir.dstu3.model.StringType;
+
+import java.util.List;
+
+public final class LookBackAnalyzer {
+
+    public static final String NO_DATA_FOR_LOOK_BACK_DETAIL = "Failed to get data for look back";
+    public static final String NO_NPI_MATCH_DETAIL = "No matches on organization npi or provider npi";
```
[the remark this answers is not in the document]
Comment (epfrankle) [reply dpc-aggregation/src/main/java/gov/cms/dpc/aggregation/service/LookBackAnalyzer.java:12]: "DPC couldn't find a claim for this MBI from an NPI in this organization" ?

File: dpc-aggregation/src/main/java/gov/cms/dpc/aggregation/service/LookBackAnalyzer.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    public static final String NO_MATCHES_DETAIL = "No matches within date limit, organization npi, and provider npi";
+    public static final String NO_DATE_MATCH_DETAIL = "No matches within look back date limit";
```
[the remark this answers is not in the document]
Comment (epfrankle) [reply dpc-aggregation/src/main/java/gov/cms/dpc/aggregation/service/LookBackAnalyzer.java:14]: "DPC couldn't find a claim within the past 18 months for this MBI from an NPI in this organization" ?

File: dpc-aggregation/src/main/java/gov/cms/dpc/aggregation/service/LookBackAnalyzer.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+import org.hl7.fhir.dstu3.model.OperationOutcome;
+import org.hl7.fhir.dstu3.model.StringType;
+
+import java.util.List;
+
+public final class LookBackAnalyzer {
+
+    public static final String NO_DATA_FOR_LOOK_BACK_DETAIL = "Failed to get data for look back";
+    public static final String NO_NPI_MATCH_DETAIL = "No matches on organization npi or provider npi";
+    public static final String NO_MATCHES_DETAIL = "No matches within date limit, organization npi, and provider npi";
```
[the remark this answers is not in the document]
Comment (epfrankle) [reply dpc-aggregation/src/main/java/gov/cms/dpc/aggregation/service/LookBackAnalyzer.java:13]: "DPC couldn't find a claim for this MBI related to an NPI in this organization, or within the past 18 months."

**Review by embh — APPROVED** [review]
🎸

**Pull request merged**